### PR TITLE
Add parallel option to updateCutoffs

### PR DESCRIPTION
**Pull request by mooreryan**
## New features

* Added option for multiple cores in the `updateCutoffs.propr` function
* Added a couple of checks to see if the user has `parallel` installed and attached, with messages as appropriate
* Added fallback to single core if `parallel` isn't available
* Updated the `updateCutoffs` function documentation.

## Tests 

* Moved the original function into `9-deprecated.R` so it can be used to test the new function
* Added `testthat` tests to test that the new version matches the old version
* Added `testthat` tests to make sure the new version matches itself with 1 and >1 cores
* Added `testthat` tests to check some of the message conditions

### CODE REVIEW

**Comment by mooreryan** [comment]
Here is a little plot showing how runtime and cores are related for different matrix sizes.

The `num_rows` ranges from 25 to 100, and the `num_cols` ranges from 250 to 1000. 

Screen Shot 2019-07-16 at 15 25 16

File: R/2-proprCall.R [hunk 1]
```diff
@@ -177,31 +193,53 @@ updateCutoffs.propr <- function(object, cutoff = seq(.05, .95, .3)){
...
-    for(cut in 1:nrow(FDR)){ # randcounts as cumsum
-
-      # Count positives as rho > cutoff, cor > cutoff, phi < cutoff, phs < cutoff
-      if(object@metric == "rho" | object@metric == "cor"){
-        FDR[cut, "randcounts"] <- FDR[cut, "randcounts"] + sum(pkt > FDR[cut, "cutoff"])
-      }else{ # phi & phs
-        FDR[cut, "randcounts"] <- FDR[cut, "randcounts"] + sum(pkt < FDR[cut, "cutoff"])
-      }
+  # Check for parallel package
+  parallelInstalled <- requireNamespace("parallel", quietly = TRUE)
```
Comment (mooreryan) [inline R/2-proprCall.R]: I just realized that `parallel` should be included with `R >= 2.14.0`.  So maybe this check isn't necessary?

**Comment by tpq** [comment]
Thanks Ryan, I'm merging this now. I'm doing it manually because (a) I am curious to see how the code works, and (b) I want to ensure compliance with CRAN policies. Thanks especially for the unit test which is well written and effective.

If you give me your e-mail, I'm happy to list you as a "ctb" in the Description. Sure it's mostly symbolic, but you can use it as evidence for OSS contributions in a c.v. or interview.

**Comment by tpq** [comment]
Now implemented in propr version 4.1.8

**Pull request closed without merging**